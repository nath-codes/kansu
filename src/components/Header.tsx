import { Link } from 'gatsby'
import { rem } from 'polished'
import React from 'react'
import KansuLogo from '../images/kansu-logo.svg'
import styled from 'styled-components'
import { Links } from './Links'

export const Header = () => (
  <Container>
    <Heading>
      <StyledLink to="/">
        <img src={KansuLogo} alt="Kansu logo" />
      </StyledLink>
    </Heading>
    <Links />
  </Container>
)

const Container = styled.header`
  align-items: center;
  display: flex;
  justify-content: space-between;
`

const Heading = styled.h1`
  margin: 0;
  color ${(props) => props.theme.colors.white}
`

const StyledLink = styled(Link)`
  display: block;
  height: ${rem(54)};
  margin-right: ${rem(24)};
`
