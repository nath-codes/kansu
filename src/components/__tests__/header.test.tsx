import 'jest-styled-components'
import React from 'react'

import Header from '../header'
import { renderWithTheme } from './helpers'

describe('Header', () => {
  it('renders correctly', () => {
    let tree = renderWithTheme(
      <Header
        logo="foo.svg"
        isMobileNavActive={false}
        handleMobileNavToggle={() => 'foo'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderWithTheme(
      <Header
        logo="foo.svg"
        isMobileNavActive={true}
        handleMobileNavToggle={() => 'foo'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
