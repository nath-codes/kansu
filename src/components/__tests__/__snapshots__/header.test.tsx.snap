// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 2.25rem;
}

.c5 {
  background-color: #ffffff;
  height: 0.25rem;
  position: relative;
  width: 100%;
  -webkit-transition-duration: 0.075s;
  transition-duration: 0.075s;
  -webkit-transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
  transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
}

.is-active .c5 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-transition-delay: 0.12s;
  transition-delay: 0.12s;
  -webkit-transition-timing-function: cubic-bezier(0.215,0.61,0.355,1);
  transition-timing-function: cubic-bezier(0.215,0.61,0.355,1);
}

.c5:before {
  border-top: solid 0.25rem #ffffff;
  content: '';
  height: 0.25rem;
  left: 0;
  opacity: 1;
  position: absolute;
  right: 0;
  top: -0.625rem;
  -webkit-transition: top 0.075s 0.12s ease,opacity 0.075s ease;
  transition: top 0.075s 0.12s ease,opacity 0.075s ease;
}

.is-active .c5:before {
  opacity: 0;
  top: 0;
  -webkit-transition: top 0.075s ease,opacity 0.075s 0.12s ease;
  transition: top 0.075s ease,opacity 0.075s 0.12s ease;
}

.c5:after {
  bottom: -0.625rem;
  border-bottom: solid 0.25rem #ffffff;
  content: '';
  height: 0.25rem;
  left: 0;
  position: absolute;
  right: 0;
  opacity: 1;
  -webkit-transition: bottom 0.075s 0.12s ease, -webkit-transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
  -webkit-transition: bottom 0.075s 0.12s ease, transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
  transition: bottom 0.075s 0.12s ease, transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
}

.is-active .c5:after {
  bottom: 0;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: bottom 0.075s ease, -webkit-transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
  -webkit-transition: bottom 0.075s ease, transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
  transition: bottom 0.075s ease, transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
}

.c3 {
  height: 3.375rem;
}

.c9 {
  color: #49555e;
  display: block;
  font-family: "Roboto",sans-serif;
  font-weight: bold;
  font-size: 1rem;
  -webkit-letter-spacing: 2.3px;
  -moz-letter-spacing: 2.3px;
  -ms-letter-spacing: 2.3px;
  letter-spacing: 2.3px;
  position: relative;
  text-transform: uppercase;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c9:hover {
  color: #ffffff;
}

.c9.active {
  color: #ffffff;
  position: relative;
}

.c9.active:after {
  border-bottom: solid 0.375rem #ffffff;
  bottom: -0.375rem;
  content: '';
  left: 0;
  position: absolute;
  right: 0;
}

.c6 {
  display: none;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  margin-right: 1.5rem;
}

.c8:last-child {
  margin-right: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  margin: 0;
}

.c2 {
  display: block;
  margin-right: 1.5rem;
}

@media only screen and (min-width:768px) {
  .c4 {
    display: none;
  }
}

@media only screen and (min-width:980px) {
  .c9.active:after {
    bottom: -0.75rem;
  }
}

@media only screen and (min-width:768px) {
  .c6 {
    display: block;
    font-family: "Roboto",sans-serif;
  }
}

<header
  className="c0"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <h1
    className="c1"
  >
    <a
      className="c2"
      href="/"
    >
      <div
        className="c3"
      >
        <img
          alt="Kansu logo"
          src="foo.svg"
        />
      </div>
    </a>
  </h1>
  <button
    aria-label="Toggle Menu"
    className="c4"
    onClick={[Function]}
  >
    <div
      className="c5"
    />
  </button>
  <nav
    className="c6"
  >
    <ul
      className="c7"
    >
      <li
        className="c8"
      >
        <a
          activeClassName="active"
          aria-label="Home page"
          className="c9"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          activeClassName="active"
          aria-label="About page"
          className="c9"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          activeClassName="active"
          aria-label="Clients page"
          className="c9"
          href="/clients"
        >
          Clients
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          activeClassName="active"
          aria-label="Contact page"
          className="c9"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Header renders correctly 2`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 2.25rem;
}

.c5 {
  background-color: #ffffff;
  height: 0.25rem;
  position: relative;
  width: 100%;
  -webkit-transition-duration: 0.075s;
  transition-duration: 0.075s;
  -webkit-transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
  transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
}

.is-active .c5 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-transition-delay: 0.12s;
  transition-delay: 0.12s;
  -webkit-transition-timing-function: cubic-bezier(0.215,0.61,0.355,1);
  transition-timing-function: cubic-bezier(0.215,0.61,0.355,1);
}

.c5:before {
  border-top: solid 0.25rem #ffffff;
  content: '';
  height: 0.25rem;
  left: 0;
  opacity: 1;
  position: absolute;
  right: 0;
  top: -0.625rem;
  -webkit-transition: top 0.075s 0.12s ease,opacity 0.075s ease;
  transition: top 0.075s 0.12s ease,opacity 0.075s ease;
}

.is-active .c5:before {
  opacity: 0;
  top: 0;
  -webkit-transition: top 0.075s ease,opacity 0.075s 0.12s ease;
  transition: top 0.075s ease,opacity 0.075s 0.12s ease;
}

.c5:after {
  bottom: -0.625rem;
  border-bottom: solid 0.25rem #ffffff;
  content: '';
  height: 0.25rem;
  left: 0;
  position: absolute;
  right: 0;
  opacity: 1;
  -webkit-transition: bottom 0.075s 0.12s ease, -webkit-transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
  -webkit-transition: bottom 0.075s 0.12s ease, transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
  transition: bottom 0.075s 0.12s ease, transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
}

.is-active .c5:after {
  bottom: 0;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: bottom 0.075s ease, -webkit-transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
  -webkit-transition: bottom 0.075s ease, transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
  transition: bottom 0.075s ease, transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
}

.c3 {
  height: 3.375rem;
}

.c12 {
  color: #49555e;
  display: block;
  font-family: "Roboto",sans-serif;
  font-weight: bold;
  font-size: 1rem;
  -webkit-letter-spacing: 2.3px;
  -moz-letter-spacing: 2.3px;
  -ms-letter-spacing: 2.3px;
  letter-spacing: 2.3px;
  position: relative;
  text-transform: uppercase;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c12:hover {
  color: #ffffff;
}

.c12.active {
  color: #ffffff;
  position: relative;
}

.c12.active:after {
  border-bottom: solid 0.375rem #ffffff;
  bottom: -0.375rem;
  content: '';
  left: 0;
  position: absolute;
  right: 0;
}

.c9 {
  display: none;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  margin-right: 1.5rem;
}

.c11:last-child {
  margin-right: 0;
}

.c6 {
  background: #00000099;
  bottom: 0;
  font-family: "Roboto",sans-serif;
  left: 0;
  position: absolute;
  right: 0;
  top: 4.5rem;
  z-index: 999;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  color: #ffffff;
  display: block;
  font-family: "Roboto",sans-serif;
  font-weight: bold;
  font-size: 1.5rem;
  -webkit-letter-spacing: 2.3px;
  -moz-letter-spacing: 2.3px;
  -ms-letter-spacing: 2.3px;
  letter-spacing: 2.3px;
  padding: 1.125rem;
  text-transform: uppercase;
}

.c8.active {
  background: #ffffff;
  color: #000000;
}

.c8.active:after {
  color: #000000;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  margin: 0;
}

.c2 {
  display: block;
  margin-right: 1.5rem;
}

@media only screen and (min-width:768px) {
  .c4 {
    display: none;
  }
}

@media only screen and (min-width:980px) {
  .c12.active:after {
    bottom: -0.75rem;
  }
}

@media only screen and (min-width:768px) {
  .c9 {
    display: block;
    font-family: "Roboto",sans-serif;
  }
}

@media only screen and (min-width:768px) {
  .c6 {
    display: none;
  }
}

<header
  className="c0"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <h1
    className="c1"
  >
    <a
      className="c2"
      href="/"
    >
      <div
        className="c3"
      >
        <img
          alt="Kansu logo"
          src="foo.svg"
        />
      </div>
    </a>
  </h1>
  <button
    aria-label="Toggle Menu"
    className="c4 is-active"
    onClick={[Function]}
  >
    <div
      className="c5"
    />
  </button>
  <nav
    className="c6"
  >
    <ul
      className="c7"
    >
      <li
        className=""
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <a
          activeClassName="active"
          className="c8"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className=""
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <a
          activeClassName="active"
          className="c8"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li
        className=""
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <a
          activeClassName="active"
          className="c8"
          href="/clients"
          onClick={[Function]}
        >
          Clients
        </a>
      </li>
      <li
        className=""
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <a
          activeClassName="active"
          className="c8"
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
  <nav
    className="c9"
  >
    <ul
      className="c10"
    >
      <li
        className="c11"
      >
        <a
          activeClassName="active"
          aria-label="Home page"
          className="c12"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="c11"
      >
        <a
          activeClassName="active"
          aria-label="About page"
          className="c12"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        className="c11"
      >
        <a
          activeClassName="active"
          aria-label="Clients page"
          className="c12"
          href="/clients"
        >
          Clients
        </a>
      </li>
      <li
        className="c11"
      >
        <a
          activeClassName="active"
          aria-label="Contact page"
          className="c12"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
